.github/workflows/release.yml

```diff
@@ -15,33 +15,158 @@
 name: 'release'
 
 on:
-  push:
-    tags:
-      - 'v*'
+  workflow_run:
```

Comment:
Why workflow run and not embedding in this workflow via workflow call. This diverges from existing prior art in jvs. I'm not opposed just want to understand the change.

Reply:
The `ci` workflow runs on every push to `main`, if we also run the `release` workflow on pushes to `main` then when we do a release the two workflows will run concurrently. 

if we embed the `ci` worflow in the `release` workflow then both `ci` and `release` will be doing the "same work". There also may or may not be a race condition with the two workflows attempting to push a tag for the same commit sha. 

Since all we care about for the release process is that the image gets built with the head commit sha, I figured it made sense to just wait for the `ci` workflow to complete since it is already doing that work. Then the `release` workflow can just re-use the image the `ci` workflow built.

Reply:
Do we want to release on every push to main though? I think we only want to release when a tag is generated, not on every push to main. When using the create-release process, a tag is automatically generated by github when the release is created, which should then kick off the release workflow instead.

Reply:
If you look at https://github.com/abcxyz/jvs/blob/main/.github/workflows/release.yml it runs on every push to main and checks that the head commit message matches a certain prefix, if not it aborts the workflow. Also, in the new process the `release` workflow is what creates the release tag so we can't trigger it on the release tag being created.

Reply:
Missed that, thanks!

Reply:
I'm confused why we're doing something different than what the JVS does?

Reply:
if we embed the `ci` workflow inside the `release` workflow and trigger the `release` workflow on pushes to main, then both workflows will run when we do a release and both workflows will build the image and tag it  with the head sha. I was worried this might result in some kind of race condition with pushing the images.

It seemed simpler to just have the `release` workflow trigger after the `ci` workflow as that way the image will already be built and we can just use it inside the `release` workflow`.